### PR TITLE
mgr/dashboard: Improve accessibility for navigation components

### PR DESCRIPTION
**Pull request by nSedrickm**
mgr/dashboard: Improve  accessibility for navigation components

* Fix all dropdown accessibility issues by setting their roles as buttons
* Increase navlink font-size to 16px
* Add new primary color variable $primary-wcag-aa-large-text: #25828e  to meet WCAG level AA color contrast ratio for active navlink backgrounds
* Add a11y e2e test for navigation component
* Log a11y error to test log
* Add custom cypress command for checking accessibility

Fixes: https://tracker.ceph.com/issues/56021
Signed-off-by: nsedrickm <nsedrick101@gmail.com>

Before
image

After
image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
@nSedrickm can you include before/after screenshots of the visual changes?

Also, for clarity, please include a short description about the changes you made in the commit(s) and the PR description. See https://github.com/ceph/ceph/commit/fab6f37052a527ec710d8d6ea48eaadd05aa72d7 as an example. This is not a requirement, but it makes it easier on the reviewers, especially reviewers who are not Ernesto or me. You could do this as a short bulleted list of all the changes you made, i.e.:

- Added titles and roles to navigation elements to meet level A accessibility guidelines
- Changed the color of X to meet level A accessibility guidelines

**Comment by nSedrickm** [comment]
Hola! here are some before and after images

Before
image

After
image

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Review by epuertat — APPROVED** [review]
LGMT @nSedrickm ! I left some comments over there.

One thing that it'd be interesting to explore is taking screenshots or accessibility rates from some tool, so after every PR merge we can see how the a11y rate of the Dashboard increases (cost vs. benefit). Have you checked if there's some automated tool that can evaluate, for example, the login page or the landing page, and rate it?

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/administration/administration.component.html [hunk 1]
```diff
@@ -4,7 +4,8 @@
   <a ngbDropdownToggle
      class="dropdown-toggle"
      i18n-title
-     title="Dashboard Settings">
+     title="Dashboard Settings"
+     role="button">
```
[the remark this answers is not in the document]
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/administration/administration.component.html:8]: Very! :)

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss [hunk 2]
```diff
@@ -190,14 +190,14 @@ $sidebar-width: 200px;
       li a {
         color: vv.$white;
         display: block;
-        font-size: 1.1em;
+        font-size: 16px;
```
[the remark this answers is not in the document]
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:193]: Sure, relative units are prefered. I used px because after using the em values I converted [here](http://pxtoem.com/) the size didn't change much. Bad calculation :)

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/language-selector/language-selector.component.html [hunk 3]
```diff
@@ -3,12 +3,17 @@
      placement="bottom-right">
   <a ngbDropdownToggle
      i18n-title
-     title="Select a Language">
+     id="toggle-button"
```
[the remark this answers is not in the document]
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/components/language-selector/language-selector.component.html:6]: Seems angular's "unique" magic only applies to classes then :)

**Comment by nSedrickm** [comment]
> LGMT @nSedrickm ! I left some comments over there.
> 
> One thing that it'd be interesting to explore is taking screenshots or accessibility rates from some tool, so after every PR merge we can see how the a11y rate of the Dashboard increases (cost vs. benefit). Have you checked if there's some automated tool that can evaluate, for example, the login page or the landing page, and rate it?

@epuertat I think [lighthouse](https://github.com/GoogleChrome/lighthouse) in chrome dev tools would be great for this. I also use [accessibility insights](https://accessibilityinsights.io/) and [axe Dev Tools](https://www.deque.com/axe/devtools/)

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
jenkins test windows

**Review by ljflores — APPROVED** [review]
Looks great Sedrick!

@epuertat what kind of QA does the dashboard usually do for just frontend changes like this?

**Comment by epuertat** [comment]
> @epuertat what kind of QA does the dashboard usually do for just frontend changes like this?

@ljflores if you mean manual, just for the submitter to provide B&A screenshots. Regarding automated ones: we have the linting for HTML/CSS files, the e2e tests (which don't perform much visual inspection, besides the Applitools/Visual Regression Testing) and the cephadm ones (which don't check the visual appearance either).

And now @nSedrickm has also added, in another PR, the axe/a11y testing into the dashboard e2e suite. So probably that other PR should be merged first, and then retrigger the e2e here, to check that this passes the new a11y test.

**Comment by ljflores** [comment]
> @ljflores if you mean manual, just for the submitter to provide B&A screenshots. Regarding automated ones: we have the linting for HTML/CSS files, the e2e tests (which don't perform much visual inspection, besides the Applitools/Visual Regression Testing) and the cephadm ones (which don't check the visual appearance either).
> 
> And now @nSedrickm has also added, in another PR, the axe/a11y testing into the dashboard e2e suite. So probably that other PR should be merged first, and then retrigger the e2e here, to check that this passes the new a11y test.

@epuertat I see, that would be #46510 correct?
@nSedrickm in this case, we will wait until your new e2e tests are merged, and then we should retrigger that test here with `jenkins test dashboard cephadm`.

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
> > @ljflores if you mean manual, just for the submitter to provide B&A screenshots. Regarding automated ones: we have the linting for HTML/CSS files, the e2e tests (which don't perform much visual inspection, besides the Applitools/Visual Regression Testing) and the cephadm ones (which don't check the visual appearance either).
> > And now @nSedrickm has also added, in another PR, the axe/a11y testing into the dashboard e2e suite. So probably that other PR should be merged first, and then retrigger the e2e here, to check that this passes the new a11y test.
> 
> @epuertat I see, that would be #46510 correct? @nSedrickm in this case, we will wait until your new e2e tests are merged, and then we should retrigger that test here with `jenkins test dashboard cephadm`.

Hi Laura, Ernesto. the e2e tests in #46510 are only for the login page. Would it be preferable to extend it to run it on each page.  If we do so it will pass for the login page since it is ready but fail for others

**Review by nizamial09 — APPROVED** [review]
LGTM! 

But the font of the sidebar being bigger seems a bit weird for me because the other texts were smaller compared to this one. Maybe we should consider increasing the font size all over?  @epuertat 

If we are following patternfly then they have [a fixed font-size of 16px for the default texts](https://www.patternfly.org/v4/guidelines/typography). We can also switch to that I guess. Only thing to consider will be the datatables (that's going to look weird-er)

**Comment by epuertat** [comment]
> @epuertat added a logger which logs accessibility errors when axe detects them. Used console.table to make the logs look presentable like the example [here](https://github.com/component-driven/cypress-axe/blob/master/terminal_output_example.png?rgh-link-date=2022-07-18T18%3A07%3A31Z) the only difference is that we replaced the node count with the actual html element for reference
> 
> image

This looks great @nSedrickm ! 🥳 

Is this output per-test or as a summary for all tests? I miss a column indicating the file/component/test where this is happening, so that devels can quickly jump into the affected file and fix it. Is that possible?

BTW is there such an `npm run a11y fix` command that automatically fixes (most of) the a11y errors, as we have with other test tools/linters?

**Comment by nSedrickm** [comment]
> > LGTM!
> > But the font of the sidebar being bigger seems a bit weird for me because the other texts were smaller compared to this one. Maybe we should consider increasing the font size all over? @epuertat
> > If we are following patternfly then they have [a fixed font-size of 16px for the default texts](https://www.patternfly.org/v4/guidelines/typography). We can also switch to that I guess. Only thing to consider will be the datatables (that's going to look weird-er)
> 
> You're right, @nizamial09. I thought Patternfly only used 16pt fonts for the navigation, but [it's for everything](https://www.patternfly.org/v4/components/navigation/html-demos/default-nav/) except for the breadcrumbs (14 pt) and other smaller texts.
> 
> So, @nSedrickm what about increasing the base text to 16pt instead? Probably that will also make the Dashboard more readable in general.

Sure, will make an update

**Comment by nSedrickm** [comment]
> > @epuertat added a logger which logs accessibility errors when axe detects them. Used console.table to make the logs look presentable like the example [here](https://github.com/component-driven/cypress-axe/blob/master/terminal_output_example.png?rgh-link-date=2022-07-18T18%3A07%3A31Z) the only difference is that we replaced the node count with the actual html element for reference
> > image
> 
> This looks great @nSedrickm ! partying_face
> 
> Is this output per-test or as a summary for all tests? I miss a column indicating the file/component/test where this is happening, so that devels can quickly jump into the affected file and fix it. Is that possible?
> 
> BTW is there such an `npm run a11y fix` command that automatically fixes (most of) the a11y errors, as we have with other test tools/linters?

Both, each error is logged at the test level and also grouped when all the tests are completed

For now no a11y linter has been implemented but hoping to add one soon. ##46972 would be a nice test case

**Comment by nSedrickm** [comment]
> LGTM!
> 
> But the font of the sidebar being bigger seems a bit weird for me because the other texts were smaller compared to this one. Maybe we should consider increasing the font size all over? @epuertat
> 
> If we are following patternfly then they have [a fixed font-size of 16px for the default texts](https://www.patternfly.org/v4/guidelines/typography). We can also switch to that I guess. Only thing to consider will be the datatables (that's going to look weird-er)

Hi @nizamial09 the default font-size is currently 12px. Increased font-size to 16px and it looks like this :)

image

14px looks better but not too out of place

image

**Comment by nSedrickm** [comment]
@epuertat added a new package for printing the error logs for better formatting. I found console.table couldn't format the output vertically

Table library output
image

console.table output
Table view

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss [hunk 4]
```diff
@@ -190,7 +190,7 @@ $sidebar-width: 200px;
       li a {
         color: vv.$white;
         display: block;
-        font-size: 1.3em;
+        font-size: 1.1em;
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:193]: @nSedrickm shouldn't it be 1rem? 1rem = 16px.

**Review by nizamial09 — APPROVED** [review]
Mostly looks good to me @nSedrickm ! Nice work with the fonts too.

Although increasing the font-size can cause some small issues in the UI like the following.

The `download and copy-to-clipboard` button goes out of place.
logs

Some input fields in the tables became smaller (or refuses to grow up :disappointed: )
tables

the ceph logo in the navbar is also smaller. Right now its height is `25px`, maybe a 30 or a 35px can save it.
ceph_logo

This is more like a concern probably but the notification sidebar icons looks too big
notification

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss (hunk 4 above)
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:193]: Thanks @nizamial09. The previous units were em so I just maintained

**Comment by nizamial09** [comment]
@nSedrickm In continuation of the discussion you raised on the standup where `cy.task()` is executing twice and it resulting into two outputs, its because there is [retry of one specifiied in our cypress.json](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/cypress.json#L21), which means that the test will retry itself once more when it is failed. So if we don't want to do that, we need to override the[ test retries in each test files](https://docs.cypress.io/guides/guides/test-retries). 

if you add a `retries` parameter in the test itself like this it will override the default retry.
```
describe('Navigation accessibility', {retries: 0 }, () => {
```
But the problem is we'll have to do this in every test suite that we need. If there is someway we can just override the retries for every tests under the `integration/a11y` then that'll be even better.

**Comment by nizamial09** [comment]
@nSedrickm btw the fonts on the landing page cards look a bit bigger to me. not sure what do you think?

**Comment by nSedrickm** [comment]
> @nSedrickm In continuation of the discussion you raised on the standup where `cy.task()` is executing twice and it resulting into two outputs, its because there is retry of 1 specifiied in our cypress.json.
> 
> https://github.com/ceph/ceph/blob/e9d361f62191a104642c85e7178d2fb4a06ac2d9/src/pybind/mgr/dashboard/frontend/cypress.json#L21
> 
> 
> which means that the test will retry itself once more when it is failed. So if we don't want to do that, we need to override the[ test retries in each test files](https://docs.cypress.io/guides/guides/test-retries).
> if you add a `retries` parameter in the test itself like this it will override the default retry.
> 
> ```
> describe('Navigation accessibility', {retries: 0 }, () => {
> ```
> 
> But the problem is we'll have to do this in every test suite that we need. If there is someway we can just override the retries for every tests under the `integration/a11y` then that'll be even better.

Thanks @nizamial09 this will help keep the logs clean. I'll do some research, perhaps there is a way to configure once for every test

**Comment by nSedrickm** [comment]
> @nSedrickm btw the fonts on the landing page cards look a bit bigger to me. not sure what do you think?

I think so too. Perhaps the font size adds up to what the cards had before making them actually larger than 16px

**Review by epuertat — COMMENTED** [review]
I have to say that I really like the new font and the bigger font size:

image

That said, there are places where this doesn't seem to fit well. Here, in the landing page, I see 2 main issues:
* The bigger fontsize removes the 'air' between components. Now everything looks too tight (probably h4 and 24 px is too big for the card titles with the new font, so the padding becomes ignored):
image

 * Additionally, the bold/stronger typefaces need to be used wisely or they'll just lose its effect of highlighting elements and simply become like WRITING EVERYTHING IN CAPS (visually "shouting"). Sometimes you can achieve more with more subtle styling (and the Red Hat font has a really sleek 'thin' typeface):
image

* Tables look nicer too (it'd be great to clean them up and remove the borders and [make them look sleeker](https://www.patternfly.org/v4/components/table/design-guidelines#expandable-table)) this will require several tweaks (note the table filters height and placement):
image
image

So, with all that said, this PR still requires works and probably manual review and tweaks (e.g: as in tables) to ensure the new font face and sizes don't look messy.

In any case, kudos to you for the courage of attempting this! My suggestion is to leave this improvement (font face and size changes) to the bottom of the queue, and only if everything else is completed then focus on this effort.

**Comment by nSedrickm** [comment]
Thanks @epuertat. We still have plenty to do accessibility wise. I don't mind working on the fonts in another PR. Perhaps we can have this update after we are done. @nizamial09 would this be ok?

**Comment by nizamial09** [comment]
> Thanks @epuertat. We still have plenty to do accessibility wise. I don't mind working on the fonts in another PR. Perhaps we can have this update after we are done. @nizamial09 would this be ok?

I fully agree. let's focus on delivering the accessibility tasks first.

**Comment by nSedrickm** [comment]
@epuertat, @nizamial09 the font update is now in #47417

File: src/pybind/mgr/dashboard/frontend/cypress/integration/a11y/navigation.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { NavigationPageHelper } from '../ui/navigation.po';
+describe('Navigation accessibility', () => {
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/cypress/integration/a11y/navigation.e2e-spec.ts:2]: then you don't have to supply it in each test cases.
```suggestion
describe('Navigation accessibility', { retries: 0 }, () => {
```

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 6]
```diff
@@ -104,6 +104,7 @@
     "rxjs": "6.6.3",
     "simplebar-angular": "2.3.6",
     "swagger-ui": "4.12.0",
+    "table": "6.8.0",
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/package.json:107]: this is more like a `devDependency` right

**Review by nizamial09 — APPROVED** [review]
minor nits.
Its good to be merged right? @nSedrickm @epuertat

File: src/pybind/mgr/dashboard/frontend/package.json (hunk 6 above)
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/package.json:107]: Ah!  yes

File: src/pybind/mgr/dashboard/frontend/cypress/integration/a11y/navigation.e2e-spec.ts (hunk 5 above)
Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/cypress/integration/a11y/navigation.e2e-spec.ts:2]: @nizamial09  I don't quite understand what to do here. Could you please explain some more

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/cypress/integration/a11y/navigation.e2e-spec.ts:2]: yup, you put the `{retries: 0}` individually in each test cases right. like this 
`it('top-nav should have no accessibility violations', { retries: 0 }, () => {`

Instead of that, if you put it just in the `describe` part, you don't need to put the `retries` in each of the test cases. The one in the `describe` will be applied to the entire block. 
https://docs.cypress.io/guides/guides/test-retries#Test-Suite-s

Comment (nSedrickm) [reply src/pybind/mgr/dashboard/frontend/cypress/integration/a11y/navigation.e2e-spec.ts:2]: Thanks a bunch!

**Review by ljflores — APPROVED** [review]
All of this looks great!

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
jenkins test dashboard cephadm

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
jenkins test dashboard cephadm

**Comment by ljflores** [comment]
@epuertat this one has received many approvals; ready to merge?

**Comment by epuertat** [comment]
jenkins retest this please

**Comment by nSedrickm** [comment]
jenkins retest this please

**Comment by epuertat** [comment]
@nSedrickm you have enought approvals here, but you need to ensure that the tests are passing (investigating whether the failures are related or not to your changes, and fixing or otherwise dismissing them).

**Comment by epuertat** [comment]
jenkins please retest this

**Comment by ljflores** [comment]
Demo (great job on this, @nSedrickm!): https://www.youtube.com/watch?v=8ZlJMzE9RcM&list=PLgmx216AWZHLva5b3ZzdI3RkxE0TFoL-Z&index=6

Let's investigate the test failures:

1.  ceph dashboard cephadm e2e tets

```
Running:  orchestrator/workflow/06-cluster-check.e2e-spec.ts                             (6 of 10)
  Estimated: 2 minutes, 36 seconds


  when cluster creation is completed
    ✓ should redirect to dashboard landing page after cluster creation (5844ms)
    Hosts page
      ✓ should change ip address of grafana, prometheus and alertmanager (9801ms)
      ✓ should reduce the count for node-exporter
      ✓ should check if monitoring stacks are running on the root host (6823ms)
      ✓ should add one more host (7423ms)
      ✓ should have removed "_no_schedule" label (5910ms)
      ✓ should display inventory
      ✓ should display daemons
      1) should check if mon daemon is running on all hosts


  8 passing (3m)
  1 failing

  1) when cluster creation is completed
       Hosts page
         should check if mon daemon is running on all hosts:
     AssertionError: Timed out retrying after 120000ms: Expected to find content: 'mon' within the selector: 'datatable-body-row datatable-body-cell:nth-child(1)' but never did.
      at PageHelper.getTableCell (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/06-cluster-check.e2e-spec.ts:1011:23)
      at Context.eval (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/06-cluster-check.e2e-spec.ts:628:23)

-----------------------------

Running:  orchestrator/workflow/08-hosts.e2e-spec.ts                                     (8 of 10)
  Estimated: 2 minutes, 52 seconds


  Host Page
    ✓ should create rgw services (10329ms)
    1) should check if rgw daemon is running on all hosts
    2) should force maintenance and exit
    ✓ should drain, remove and add the host back (70182ms)
    ✓ should show the exact count of daemons (5217ms)


  3 passing (6m)
  2 failing

  1) Host Page
       should check if rgw daemon is running on all hosts:
     AssertionError: Timed out retrying after 120000ms: Expected to find content: 'rgw' within the selector: 'datatable-body-row datatable-body-cell:nth-child(1)' but never did.
      at PageHelper.getTableCell (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/08-hosts.e2e-spec.ts:715:23)
      at Context.eval (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/08-hosts.e2e-spec.ts:411:23)

  2) Host Page
       should force maintenance and exit:
     AssertionError: Timed out retrying after 120000ms: Expected to find element: `cd-modal`, but never found it.
      at HostsPageHelper.maintenance (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/08-hosts.e2e-spec.ts:187:16)
      at descriptor.value [as maintenance] (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/08-hosts.e2e-spec.ts:558:20)
      at Context.eval (https://192.168.100.100:8443/__cypress/tests?p=cypress/integration/orchestrator/workflow/08-hosts.e2e-spec.ts:520:15)
```

I'm not seeing how these are obviously related to your code changes, but I am seeing this check pass on other dashboard PRs. I'd recommend retriggering the test with `jenkins test dashboard cephadm` and seeing if it reproduces. If it does, it might have something to do with your code.

2. ceph dashboard tests

```
2022-08-21T18:04:28.849+0000 7f81cf9144c0 -1 bluestore(/home/jenkins-build/build/workspace/ceph-dashboard-pull-requests/build/run/1/dev/osd0/block) _read_bdev_label failed to open /home/jenkins-build/build/workspace/ceph-dashboard-pull-requests/build/run/1/dev/osd0/block: (2) No such file or directory
```

This seems unrelated to your code. Not sure exactly of the root cause, but it could be an environmental issue on the node where this check was run. Let's retrigger it with `jenkins test dashboard`.

3. ceph windows tests

This one is failing on all main PRs right now. It's definitely unrelated to your code. I think I've seen some other developers trying to fix it.

4.  make check

```
1 tests failed out of 282

Total Test time (real) = 958.71 sec

The following tests FAILED:
	 32 - run-rbd-unit-tests-61.sh (Failed)
```

This check is failing due to an RBD unit test. Your code is clearly not touching any RBD code, so it's unrelated. This is also happening on all PRs right now. We should actually look into whether this is getting tracked/handled.

5. make check arm64

```
CMake Error at /opt/ceph/lib/aarch64-linux-gnu/cmake/Boost-1.79.0/BoostConfig.cmake:141 (find_package):
```

This is a known failure that is getting addressed. You can also see it failing on other PRs.


To sum up, you should look into the two dashboard checks to make sure nothing going on there is related to your code. But everything else is unrelated.

**Comment by nSedrickm** [comment]
Thanks a lot @ljflores

**Comment by epuertat** [comment]
jenkins test make check

**Comment by epuertat** [comment]
Thanks @ljflores for the thorough analysis! Let's retrigger the `make check` (unrelated rbd failure) and tag it as ready-to-merge, so it can be merged if that one passes.

**Comment by nizamial09** [comment]
rbd failure [again](https://jenkins.ceph.com/job/ceph-pull-requests/102053/). retriggering

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by pereman2** [comment]
jenkins test dashboard cephadm

**Comment by pereman2** [comment]
jenkins test dashboard

**Comment by pereman2** [comment]
jenkins test dashboard

**Comment by pereman2** [comment]
jenkins test dashboard cephadm

**Comment by pereman2** [comment]
```
	 16 - run-tox-mgr-dashboard-openapi (Failed)
```
You can run `tox -e openapi-fix` in the dashboard folder to generate a new openapi.yaml

**Review by pereman2 — COMMENTED** [review]
```
	 16 - run-tox-mgr-dashboard-openapi (Failed)
```
You can run `tox -e openapi-fix` in the dashboard folder to generate a new openapi.yaml

**Comment by nSedrickm** [comment]
> 

Thanks @pereman2 I ran the command but tox is not found. Is it a dashboard dependency or do I install from my system package manager?

**Comment by nSedrickm** [comment]
Hi @pereman2, I generated a new openapi.yaml file but make check fails. Please, how can I fix this one?

image

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test windows

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test make check

**Comment by nSedrickm** [comment]
jenkins test dashboard cephadm

**Comment by nSedrickm** [comment]
jenkins test make check arm64

**Comment by nSedrickm** [comment]
jenkins test windows

**Comment by nSedrickm** [comment]
jenkins test make check arm64

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
@epuertat @pereman2 the dashboard test is failing from Applitools detecting visual diffs:
```
 1) Login Page
       "after all" hook for "types login credentials and takes screenshot":
     Error: Eyes-Cypress detected diffs or errors during execution of visual tests. See details at: https://eyes.applitools.com/app/batches/00000251739738939827/00000251739738939577?accountId=hCTedwnSTUWSCZlXPqua0g~~
       Passed - 4 tests
         ✓ Login Component Check [1920x1080]
         ✓ Login Component Check [1920x1080]
         ✓ Login Component Check [800x600]
         ✓ Login Component Check [800x600]
       Diffs detected - 4 tests
         ⚠ Dashboard Component [1920x1080]
         ⚠ Dashboard Component [1920x1080]
         ⚠ Dashboard Component [800x600]
         ⚠ Dashboard Component [800x600]

       See details at: https://eyes.applitools.com/app/batches/00000251739738939827/00000251739738939577?accountId=hCTedwnSTUWSCZlXPqua0g~~
```
This is expected, as Sedrick made some visual changes. It seems that accepting/saving the new changes in Applitools is not persistent, since after retriggering the test, it has failed again.

What is usually done in this situation?

**Comment by pereman2** [comment]
@ljflores @nSedrickm In applitools you can accept new changes made to the UI. If you click on an unresolved test and then hover over the image, a thumbs up icon will appear. Click on that if it is a expected change and applitools will update its screenshots.
image

**Comment by nizamial09** [comment]
applitools were failing again, but i'll merge this and approve it on the applitools main baseline branch.

**Comment by nSedrickm** [comment]
Thanks @pereman2 @nizamial09
